coretrace/determineelementintersectionnew.cpp

```diff
@@ -397,16 +416,19 @@ void DetermineElementIntersectionNew(
 				*BacksideFlag = 1;
 				
 			*Intercept = 1;
-			goto Label_100;
+			//goto Label_100;
 
 		break;
 
 	case 'a':
 	case 'A': //Annulus or torus contour
 	
-			if ( (Element->ParameterA == 0.0) && (Element->ParameterB == 0.0) ) goto Label_5; //torus
-
-			if ( (r < Element->ParameterA) || (r > Element->ParameterB) )
+			if ((Element->ParameterA == 0.0) && (Element->ParameterB == 0.0))
```

Comment:
Can and should we combine these two logic statements somehow?  Maybe before changing anything we should understand when the first condition is true? I see this is only for a torus... Fine for now. But should be address when we rewrite this to handle the new data structures.